docs/reference/deprecation-notice.md

```diff
@@ -0,0 +1,76 @@
+# Feature Deprecation Notice
+---
+
+This document outlines the features that are deprecated in the past releases. Deprecation means that these features will no longer be supported or recommended for use, and will be completely removed in the next major release. Users should transition to alternative features or solutions as specified below.
+
+## Table of contents
+- [Feature Deprecation Notice](#feature-deprecation-notice)
+  - [Table of contents](#table-of-contents)
+  - [Deprecation Summary](#deprecation-summary)
+  - [Deprecation Details](#deprecation-details)
+    - [`Name` and `Type` fields in verifierReport](#name-and-type-fields-in-verifierreport)
+    - [`CertificateStore` CRD](#certificatestore-crd)
+    - [Legacy Cosign Veification](#legacy-cosign-veification)
+    - [licensechecker verifier plugin](#licensechecker-verifier-plugin)
+
+## Deprecation Summary
+
+| Deprecated Feature                         | Deprecation Version | Removal Version | Replacement/Alternative                       |
+| ------------------------------------------ | ------------------- | --------------- | --------------------------------------------- |
+| `Name` and `Type` fields in verifierReport | v1.3.0              | v2.0.0          | Use `VerifierName` and `VerifierName` instead |
+| `CertificateStore` CRD                     | v1.2.0              | v2.0.0          | Use `KeyManagementProvider` CRD instead       |
```

Comment:
Suggestion to add linked doc for the replacement feature

Reply:
I added doc link in the detailed list if it exists. For the summary table, I want to keep it concise enough.